### PR TITLE
Various fixes

### LINKED ISSUE #284: Don't display removed message in conversation

### LINKED ISSUE #661: Attachments zone for drag and drop doesn't stay blue.
**Steps to reproduce:**
* Drag an image on attachment zone in compose page
* Zone becomes blue
* Continue to drag in the zone

**Expected behavior:** Zone stays blue 
**Observed behavior:** Blue disappears

### PR DESCRIPTION
**Pull request by jsilvestre**
* Fix #284 and a related bug found during development.
* Fix #661
* Fix regressions
    * fix react warnings
    * fix unread marker 
    * fix Icon component's class name
    * fix unexpected type for selectedMailboxID

### CODE REVIEW

File: client/app/stores/message_store.coffee [hunk 1]
```diff
@@ -300,6 +301,14 @@ class MessageStore extends Store
...
                 else
                     onReceiveRawMessage message
+
+                # Update conversation length
+                conversationID = {message}
+                currentLength = _conversationLengths.get conversationID
+                newLength = currentLength - 1
+                console.log currentLength, newLength
+                _conversationLengths = _conversationLengths
+                                                 .set(conversationID, newLength)
```
Comment (nono) [inline client/app/stores/message_store.coffee]: @jsilvestre you forgot a `console.log`. And, `get + set` is an `update`, isn't it?

Comment (jsilvestre) [inline client/app/stores/message_store.coffee]: You are right, thank you, gonna fix it!

**Comment by jsilvestre** [comment]
**Please wait before merging this PR, thanks**

**Comment by jsilvestre** [comment]
All right, you can review it.

**Pull request merged**